packages/core-api/src/app/configLoaders/composeConfigLoaders.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AppConfig } from '@backstage/config';
+import { AppConfigLoader } from '../types';
+
+/**
+ * Executes config loaders and merges the results. The precedence is determined by the order they are supplied to the
+ * function (first parameter has highest priority).
+ *
+ * Where loaderA returns [appConfigA1, appConfigA2] and loaderB returns [appConfigB]
+ * await composeConfigLoaders(loaderA, loaderB) returns [appConfigA1,appConfigA2, appConfigB]
+ */
+export const composeConfigLoaders = (
+  ...configLoaders: AppConfigLoader[]
+): AppConfigLoader => async (): Promise<AppConfig[]> => {
+  const loaderResults = await Promise.all(
+    configLoaders.map(loader => loader()),
+  );
+  return loaderResults.flat();
```

Comment:
Hmm .flat() is not natively supported in IE is that an issue? or will babel sort it out?

Reply:
It's fine! :grin: We're not aiming to support IE, if it absolutely necessary to do so then global polyfills can be added separately